tests/remotes.py

```diff
@@ -271,8 +255,21 @@ def put_objects(remote, objects):
 
 
 class Azure:
-    should_test = staticmethod(_should_test_azure)
-    get_url = staticmethod(get_azure_url)
+    @staticmethod
+    def should_test():
```

Comment:
The logic is different from other `should_test`, for example:

```python
class GDrive:
    @staticmethod
    def should_test():
        return os.getenv("GDRIVE_USER_CREDENTIALS_DATA") is not None
```

At least as credentials are set, it will run the tests.

But as I'm reading this, if `DVC_TEST_AZURE` is defined, it is assumed that `AZURE_STORAGE_{CONTAINER_NAME,CONNECTION_STRING}` are set.

Why not going for testing only the credentials?

Reply:
@mroutis Gdrive is the only one that doesn't respect the previous convention :slightly_smiling_face: This change does. `DVC_TEST*` are useful when you want to ensure that your tests will run even if the env vars are not defined, for example when you have some other way of providing credentials already. For example, should_test for s3 tests for AWS_ env vars, but I don't define them locally, since I have the default ones that work as is.

Reply:
@efiop , got it :+1: